### PR TITLE
Update the docs on passthrough proxy

### LINKED ISSUE #20070: Update passthrough behavior and docs
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

dist/quarkus

### Describe the bug

Update the docs about a `passthrough` proxy. 

### Version

21.1.1

### Expected behavior

A `passthrough` proxy is usually handling TLS and doesn't add headers.

### Anything else?

I will create a PR

### PR DESCRIPTION
**Pull request by ahus1**
Closes #20070

### CODE REVIEW

**Comment by ahus1** [comment]
@vmuzikar, @pedroigor - this implements the change we discussed. Please review and approve. 

Looking at the code, no review from the storage team is required IMHO.

File: docs/guides/server/reverseproxy.adoc [hunk 1]
```diff
@@ -21,19 +21,19 @@ reencrypt:: Requires communication through HTTPS between the proxy and Keycloak.
...
 To select the proxy mode, enter this command:
 
 <@kc.start parameters="--proxy <mode>"/>
 
 == Configure the reverse proxy
+
 Some Keycloak features rely on the assumption that the remote address of the HTTP request connecting to Keycloak is the real IP address of the clients machine.
 
-When you have a reverse proxy or a load balancer in front of Keycloak, please make sure your reverse proxy is overriding the following headers:
+When in **edge** or **reencrypt** proxy mode, Keycloak will parse the following headers expects the reverse to set them:
```
Comment (vmuzikar) [inline docs/guides/server/reverseproxy.adoc:36]:
```suggestion
When in **edge** or **reencrypt** proxy mode, Keycloak will parse the following headers and expects the reverse proxy to set them:
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Thanks, @ahus1!

I think we should also reflect this in the upgrade guide, it's a breaking change afterall. WDYT?

**Comment by ahus1** [comment]
@vmuzikar - I've now added paragraphs to the upgrade and migration guide. This is now ready for a another review. Thanks!

**Review by vmuzikar — DISMISSED** [review]
LGTM, thanks @ahus1!

**Comment by ahus1** [comment]
Rebased after conflicts.

**Review by vmuzikar — DISMISSED** [review]

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc [hunk 2]
```diff
@@ -129,4 +129,12 @@ The new artifact with Java EE support is `org.keycloak:keycloak-admin-client-jee
     <artifactId>keycloak-admin-client-jee</artifactId>
     <version>22.0.0</version>
 </dependency>
-----
\ No newline at end of file
+----
+
+= Passthrough proxy mode changes
+
+Keycloak's proxy configuration setting for mode *passthrough* does now longer parse HTTP forwarding headers in the request, as when a proxy forwards an HTTPS connection in passthrough mode, a proxy is unable to add, remove or update HTTP headers.
```
Comment (mabartos) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:136]: @ahus1 Should be like this, right?

```suggestion
Keycloak's proxy configuration setting for mode *passthrough* no longer parses HTTP forwarding headers in the request, as when a proxy forwards an HTTPS connection in passthrough mode, a proxy is unable to add, remove or update HTTP headers.
```

Comment (mabartos) [reply docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:136]: Or just `no longer parses`?

Comment (ahus1) [reply docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc]: Fixed in latest commit, ready for review.

**Review by mabartos — APPROVED** [review]
@ahus1 LGTM

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**